library/core/src/num/int_macros.rs

```diff
@@ -1670,6 +1670,152 @@ macro_rules! int_impl {
             }
         }
 
+        /// Returns the logarithm of the number with respect to an arbitrary base.
+        ///
+        /// This method may not be optimized owing to implementation details;
+        /// `log2` can produce results more efficiently for base 2, and `log10`
+        /// can produce results more efficiently for base 10.
+        ///
+        /// # Panics
+        ///
+        /// Panics if the number is negative or zero, or if the base is not at least 2.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_log)]
+        #[doc = concat!("assert_eq!(5", stringify!($SelfT), ".log(5), 1);")]
+        /// ```
+        #[unstable(feature = "int_log", issue = "70887")]
+        #[must_use = "this returns the result of the operation, \
+                        without modifying the original"]
+        #[inline]
+        pub const fn log(self, base: Self) -> Self {
```

Comment:
Thank-you for shepherding this code through the process! Please consider making the return type of the log functions u32 rather than Self. log is the inverse of pow, and pow accepts a u32 as input. When testing this code, I made a quickcheck test that asserted that log was indeed the inverse of pow. You can see the tests here: https://gist.github.com/rust-play/cee6917b15c465635d31af30f9f60599

Reply:
Do you have any response to "Please consider making the return type of the log functions u32 rather than Self. log is the inverse of pow, and pow accepts a u32 as input."? Thanks!

Reply:
Oh, that's a good point. Yeah we should probably change that!